### PR TITLE
feat(e2e): sets up playwright, 1st user flow test

### PR DESCRIPTION
**Pull request by billhimmelsbach**
Adds playwright to the test suite including deps, config, test fixtures, and sample tests. This PR gets improved upon by other pending PRs.

Closes: #709 

## Changes

(in order of [files changed](https://github.com/cfpb/sbl-frontend/pull/776/files))

- add playwright files to `.gitignore`
- add yarn specific config to `.npmrc` for playright/vscode integration
- add yarn pnp specific config to `settings.json`
- add very basic test instructions to the `readme.md`
- provide a sample test to `example.spec.demo.ts` to show different methods available to use during testing
- add `textFixture.ts` that spins up new test accounts and provides convenience functions to navigate through the app
- add `completeFilngFlowWithOnlyWarnings.spec.ts` to test through sign and submit
- add `sampleTestCfpbLogoGoesToConsumerFinance.spec.ts` as a basic test to make sure site loads
- add `point-of-contact-data-1.json` for basic point of contact info
- add  `sbl-validations-all-pass-small.csv` as a test file
- creates utility functions to drive the fixtures
  - `createDomainAssociation.ts`
  - `createInstitution.ts`
  - `createKeycloakUser.ts`
  - `getKeycloakToken.ts`
- add `playwright.config.ts` with just the basic configuration and only using chrome for now
- change `Loading.tsx` to be a paragraph instead of a heading since we don't need to have a heading for unstructured loading content (and it also helps with all the tests)

## Notes
- lots of PRs depend on this one, so some things have been improved in later PRs
- there are typescript errors that will need to be fixed, but there's a [yarn pnp bug](https://github.com/microsoft/playwright/issues/21058) that keeps types from being easily imported. We'll fix it up in a future PR, and when we switch to node_modules, we might get the types imported easily too.
- eslint gets fixed [in a later PR](https://github.com/cfpb/sbl-frontend/pull/821), so we'll clean it up after that PR gets merged
- Right now the fixture doesn't clean up the generated users/institutions, but we're just running these tests locally for now, and we'll be having conversations about adding APIs for that on Monday
- this is just the start of more playwright awesomeness! 😎

## How to test this PR

1. Run `yarn playwright test --ui`, see if the tests pass
2. Does the site still operate normally otherwise?

## Screenshots
Screenshot 2024-07-25 at 11 04 34 AM

### CODE REVIEW

File: e2e/fixtures/testFixture.ts [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+  navigateToProvidePointOfContact: Page;
+  navigateToSignAndSubmit: Page;
+}>({
+  // allowing fixture functions to be called without being used immediately
+  // eslint-disable @typescript-eslint/no-unused-expressions
+  authHook: [
+    async ({ page }, use) => {
+      // eslint-disable @typescript-eslint/no-magic-numbers
+      const seed = Math.floor(Math.random() * 10_000_000_000).toString();
+      const testUsername = `playwright-test-user-${seed}`;
```
[the remark this answers is not in the document]
Comment (billhimmelsbach) [reply e2e/fixtures/testFixture.ts:27]: These are ephemeral usernames and passwords only on local env.

File: e2e/fixtures/testFixture.ts [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+      const testFirstName = 'Playwright';
+      const testLastName = `Test User ${seed}`;
+      const testEmailDomain = `${seed}.gov`;
+      const testUserEmail = `playwright-test-user-${seed}@${testEmailDomain}`;
+      const testUserPassword = `playwright-test-user-${seed}-password`;
```
[the remark this answers is not in the document]
Comment (billhimmelsbach) [reply e2e/fixtures/testFixture.ts:32]: These are ephemeral usernames and passwords only on local env.

File: e2e/utils/createKeycloakUser.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+}
+
+// Allow developers to disable routing in development
+export default async function createKeycloakUser({
+  testUserEmail,
+  testUsername,
+  testFirstName,
+  testLastName,
+  testUserPassword,
+}: CreateKeycloakUserProperties): Promise<void> {
```
[the remark this answers is not in the document]
Comment (billhimmelsbach) [reply e2e/utils/createKeycloakUser.ts:34]: These are ephemeral usernames and passwords only on local env.

[the remark this answers is not in the document]
Comment (billhimmelsbach) [reply e2e/utils/createKeycloakUser.ts:34]: Added a commit to change to make the linter happy.

https://github.com/cfpb/sbl-frontend/pull/776/commits/4cebe9e20b375c942b5d510e5a66306f2f1608fd

File: e2e/fixtures/testFixture.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (billhimmelsbach) [reply e2e/fixtures/testFixture.ts:34]: Added a commit to change to make the linter happy.

https://github.com/cfpb/sbl-frontend/pull/776/commits/4cebe9e20b375c942b5d510e5a66306f2f1608fd

File: e2e/fixtures/testFixture.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (billhimmelsbach) [reply e2e/fixtures/testFixture.ts:29]: Added a commit to change to make the linter happy.

https://github.com/cfpb/sbl-frontend/pull/776/commits/4cebe9e20b375c942b5d510e5a66306f2f1608fd

**Review by ojbravo — APPROVED** [review]
All playwright tests passed for me!

**Pull request merged**